lib/nfg/capistrano/tasks/deploy.rake

```diff
@@ -0,0 +1,37 @@
+require 'aws-sdk-s3'
+
+namespace :deploy do
+  Rake::Task['deploy:compile_assets'].clear
+
+  desc 'Download or compile assets'
+  task :compile_assets => [:set_rails_env] do
+    assets_cache_prefix = Nfg::Capistrano::Config.circleci[:parameters][:assets_cache_prefix][:default]
+    revision = %x(git --no-pager log -1 --pretty=format:%H)
+    branch = fetch(:branch).gsub('/', '_')
+    assets_filename="#{assets_cache_prefix}-assets-#{revision}-#{branch}.tar.gz"
+    s3_bucket = Nfg::Capistrano::Config.circleci[:parameters][:s3_bucket][:default]
+    aws_credentials = Aws::Credentials.new(
+      Nfg::Capistrano::Config.api_keys[:aws_access_key_id],
+      Nfg::Capistrano::Config.api_keys[:aws_secret_access_key]
+    )
+    s3 = Aws::S3::Client.new(
+      region: ENV.fetch('AWS_REGION', 'us-east-1'),
+      credentials: aws_credentials
+    )
+
+    begin
+      s3.head_object(bucket: s3_bucket.gsub('s3://', ''), key: "assets/#{assets_filename}")
+      on release_roles :all do
+        execute :s3cmd, "--force get #{s3_bucket}/assets/#{assets_filename} #{shared_path}/public/assets/#{assets_filename}"
```

Comment:
I get the impression shared_path is a capistrano thing but I don't see a clear definition for it documented on their website. I'm inferring what the value is from an answer here:

https://stackoverflow.com/questions/30113279/get-path-to-capistrano-shared-path-from-ruby

not a big deal, just odd

Reply:
The `shared_path` is a directory containing resources that can be shared between different releases. This would include the compiled assets, the dependencies cache, the yaml config files (database.yml etc), and also the webpacker packs directory.